src/topology/mod.rs

```diff
@@ -642,6 +651,9 @@ impl RunningTopology {
         }
 
         self.outputs.insert(name.to_string(), output);
+
+        #[cfg(feature = "api")]
+        self.reconnect_tap_sinks(name);
```

Comment:
When inputs are removed, we reconnect tap sinks here to ensure everything is wired up properly.